### ISSUE TITLE
Rule proposal: Rule to prevent `in` operator on arrays (valid but likely unintended)

### ISSUE THREAD

**Issue by f-schnabel** [typescript-eslint/typescript-eslint#11913]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [x] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I'd like to request a new ESLint rule to flag usage of the `in` operator on arrays, as it's likely to catch bugs where developers intend to check array contents but use the wrong method.
Example:
```ts
"" in []  // Returns false, but probably not what was intended
[].includes("")  // More explicit and correct
```
The `in` operator checks for object properties, not array contents. While it's valid TypeScript/JavaScript syntax, it's almost always a mistake when used on arrays. Developers typically mean to use `.includes()`, `.indexOf()`, or `.some()`.
Or otherwise only allow numbers on the left side of the `in` operator if the right side is an array.

Why this matters:
This is valid Python syntax as well, which might lead developers familiar with Python to accidentally use in on arrays in TypeScript/JavaScript without realizing it checks properties rather than contents.

Request:
Either a new rule like `prefer-array-methods-over-in` or `no-in-operator-on-arrays`, or confirmation that an existing rule should catch this and clarification on how to enable it.

### Fail Cases

```TypeScript
const list = [];
if ("hello" in list) {
  console.log("");
}
```

### Pass Cases

```TypeScript
const list = [];
if (list.includes("hello")) {
  console.log("");
}
```

### Additional Info

There also exists a similar rule https://typescript-eslint.io/rules/no-for-in-array but this only catches for loops, not if statements.

**Comment by JoshuaKGoldberg** [comment]
Duplicate of #5677. We agree, >=1 rule like this would be very useful! 🙂
